src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala

```diff
@@ -0,0 +1,248 @@
+package io.iohk.ethereum.db.dataSource
+
+import java.util.concurrent.locks.ReentrantReadWriteLock
+
+import io.iohk.ethereum.db.dataSource.RocksDbDataSource.withResources
+import org.rocksdb._
+import org.slf4j.LoggerFactory
+
+import scala.util.control.NonFatal
+
+class RocksDbDataSource(private var db: RocksDB, private val rocksDbConfig: RocksDbConfig, private var readOptions: ReadOptions) extends DataSource {
+
+  private val logger = LoggerFactory.getLogger("rocks-db")
+
+  /**
+    * This function obtains the associated value to a key, if there exists one.
+    *
+    * @param namespace which will be searched for the key.
+    * @param key       the key retrieve the value.
+    * @return the value associated with the passed key.
+    */
+  override def get(namespace: Namespace, key: Key): Option[Value] = {
+    RocksDbDataSource.dbLock.readLock().lock()
+    try {
+      Option(db.get(readOptions, (namespace ++ key).toArray))
+    } catch {
+      case e: Exception =>
```

Comment:
(comment in this line but applies to all exceptions)
I am not sure that swallowing  exceptions this way is good idea e.g it can lead to case during execution of transaction that mantis will think some account does not exist and create new one, or some update will be lost etc.
I think you should catch `case NonFatal(ex)`  (a if it its fatal error we do not do anything ), and log error, and just re throw it as `RuntimeException`, similary to ethereumj `https://github.com/ethereum/ethereumj/blob/develop/ethereumj-core/src/main/java/org/ethereum/datasource/rocksdb/RocksDbDataSource.java#L280`

Reply:
good point